tron/trondaemon.py

```diff
@@ -98,32 +98,38 @@ def __init__(self, options):
         setup_logging(self.options)
 
         self.mcp = None
-        self._sigint_handler = self._make_sigint_handler(
-            signal.getsignal(signal.SIGINT)
-        )
         self.context = self._build_context(options)
         self.manhole_sock = f"{self.options.working_dir}/manhole.sock"
 
     def _build_context(self, options):
         return NoDaemonContext(
             lockfile=flockfile.FlockFile(options.lock_file),
             working_directory=options.working_dir,
-            signal_map={
-                signal.SIGHUP: signal.SIG_DFL,
-                signal.SIGINT: signal.default_int_handler,
-                signal.SIGTERM: signal.SIG_DFL,
-                signal.SIGQUIT: signal.SIG_DFL,
-                signal.SIGUSR1: signal.SIG_DFL,
-            },
+            signal_map={signal.SIGINT: lambda sig, frame: sig},
         )
 
     def run(self):
         with self.context:
+            signal_map = {
+                signal.SIGHUP: self._handle_reconfigure,
+                signal.SIGINT: self._handle_shutdown,
+                signal.SIGTERM: self._handle_shutdown,
+                signal.SIGQUIT: self._handle_shutdown,
+                signal.SIGUSR1: self._handle_debug,
+            }
+            signal.pthread_sigmask(signal.SIG_BLOCK, signal_map.keys())
+
             self._run_mcp()
             self._run_www_api()
             self._run_manhole()
             self._run_reactor()
 
+            while True:
+                signum = signal.sigwait(list(signal_map.keys()))
```

Comment:
@keymone Do you think we still need to add a sleep or timeout here?

Reply:
Never mind, we may have needed a timeout with sigtimedwait, but we shouldn't need one for sigwait, which suspends the thread until a signal is available.